service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java

```diff
@@ -642,26 +660,84 @@ SITE_ENTRIES.COMMON_NAME, USERS.FIRST_NAME, USERS.LAST_NAME, USERS.EMAIL, coales
         .having(stewardshipCondition)
         .fetch();
 
-    return records.stream().map(rec -> {
-      String adopterName = rec.get(USERS.FIRST_NAME) + ' ' + rec.get(USERS.LAST_NAME);
-      Integer lastActivityWeeks = rec.get(STEWARDSHIP.PERFORMED_ON) != null
-          ? (int) ChronoUnit.WEEKS.between(rec.get(STEWARDSHIP.PERFORMED_ON).toLocalDate(), LocalDate.now())
-          : null;
-      String dateAdopted = rec.get(ADOPTED_SITES.DATE_ADOPTED) != null
-          ? rec.get(ADOPTED_SITES.DATE_ADOPTED).toString()
-          : "";
-
-      return new FilterSitesResponse(
-          rec.get(SITES.ID),
-          rec.get(SITES.ADDRESS),
-          rec.get(ADOPTED_SITES.USER_ID),
-          adopterName,
-          rec.get(USERS.EMAIL),
-          dateAdopted,
-          rec.get(ACTIVITY_COUNT_COLUMN, Integer.class),
-          rec.get(SITES.NEIGHBORHOOD_ID),
-          lastActivityWeeks
-      );
-    }).collect(Collectors.toList());
+    return records.stream()
+        .map(
+            rec -> {
+              String adopterName = rec.get(USERS.FIRST_NAME) + ' ' + rec.get(USERS.LAST_NAME);
+              Integer lastActivityWeeks =
+                  rec.get(STEWARDSHIP.PERFORMED_ON) != null
+                      ? (int)
+                          ChronoUnit.WEEKS.between(
+                              rec.get(STEWARDSHIP.PERFORMED_ON).toLocalDate(), LocalDate.now())
+                      : null;
+              String dateAdopted =
+                  rec.get(ADOPTED_SITES.DATE_ADOPTED) != null
+                      ? rec.get(ADOPTED_SITES.DATE_ADOPTED).toString()
+                      : "";
+
+              return new FilterSitesResponse(
+                  rec.get(SITES.ID),
+                  rec.get(SITES.ADDRESS),
+                  rec.get(ADOPTED_SITES.USER_ID),
+                  adopterName,
+                  rec.get(USERS.EMAIL),
+                  dateAdopted,
+                  rec.get(ACTIVITY_COUNT_COLUMN, Integer.class),
+                  rec.get(SITES.NEIGHBORHOOD_ID),
+                  lastActivityWeeks);
+            })
+        .collect(Collectors.toList());
+  }
+
+  public void editSiteEntry(JWTData userData, int entryId, UpdateSiteRequest editSiteEntryRequest) {
+    checkEntryExists(entryId);
+    assertAdminOrSuperAdmin(userData.getPrivilegeLevel());
+
+    SiteEntriesRecord siteEntriesRecord =
+        db.selectFrom(SITE_ENTRIES).where(SITE_ENTRIES.ID.eq(entryId)).fetchOne();
+
+    siteEntriesRecord.setUserId(userData.getUserId());
+    siteEntriesRecord.setUpdatedAt(new Timestamp(System.currentTimeMillis()));
+    siteEntriesRecord.setTreePresent(editSiteEntryRequest.isTreePresent());
```

Comment:
⛏️  This giant block of code where we set all of the site entry fields already exists in two other places - can we pull those chunks out into its own method (and call it `populateSiteEntry` or something)? That way, if we ever add a new column to this table, we'll just need to change the one place where we fill in site entry data.

Reply:
Thanks for the tips Avery! I made a PR for the docs and refactored the duplicate code into the helper.